app/src/main/java/com/infomaniak/drive/data/cache/FileController.kt

```diff
@@ -822,15 +822,8 @@ object FileController {
     private fun FileActivity.applyFileActivity(realm: Realm, returnResponse: ArrayMap<Int, FileActivity>, currentFolder: File) {
         when (getAction()) {
             FileActivityType.FILE_DELETE,
-            FileActivityType.FILE_MOVE_OUT,
-            FileActivityType.FILE_TRASH -> {
-                if (returnResponse[fileId] == null || returnResponse[fileId]?.createdAt?.time == createdAt.time) { // Api fix
-                    getParentFile(fileId = fileId, realm = realm)?.let { parent ->
-                        if (parent.id == currentFolder.id) removeFile(fileId, customRealm = realm, recursive = false)
-                    }
-                    returnResponse[fileId] = this
-                }
-            }
+            FileActivityType.FILE_TRASH -> removeFileWhenEnteringFolder(true, realm, returnResponse, currentFolder)
+            FileActivityType.FILE_MOVE_OUT -> removeFileWhenEnteringFolder(false, realm, returnResponse, currentFolder)
```

Comment:
The name is too big, and I think we can largely simplify it, here, for example a name like `handleRemoveAction` is more simple